src/ui/public/error_auto_create_index/error_auto_create_index.test.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+jest.mock('../chrome', () => ({
+  addBasePath: path => `myBase/${path}`,
+}));
+jest.mock('../metadata', () => ({
+  metadata: {
+    version: 'my-version',
+  },
+}));
+
+import fetchMock from 'fetch-mock';
+import { kfetch } from 'ui/kfetch';
+import { isAutoCreateIndexError } from './error_auto_create_index';
+
+describe('isAutoCreateIndexError correctly handles FetchError thrown by kfetch', () => {
+  describe('404', () => {
+    beforeEach(() => {
+      fetchMock.post({
+        matcher: '*',
+        response: {
+          status: 404,
+        },
+      });
+    });
+    afterEach(() => fetchMock.restore());
+
+    test('should return false', async () => {
+      let gotError = false;
```

Comment:
@nreese stumbled over this while updating `kfetch`. `expect.assertions(1)` is very handy is this exact scenario (https://jest-bot.github.io/jest/docs/expect.html#expectassertionsnumber)